lib/wlib/memory/Memory.h

```diff
@@ -53,85 +55,172 @@ static MemoryInitDestroy __g_smemoryInitDestroy;
  */
 void *__memory_alloc(uint32_t size, bool anObject);
 
-/**
- * Used internally not for client's use
- * Use the @code realloc @endcode function
- */
-void *__memory_realloc(void *ptr, uint32_t size);
-
 /**
  * Used internally not for client's use
  * Use the @code free @endcode function
  */
 void __memory_free(void *ptr);
 
 /**
- * This allocates memory of the size provided. Memory allocated could be greater than what has been
- * asked in order to accommodate fixed memory allocations. If there is not a sufficient memory
+ * This allocates memory of the size of the type provided. Memory allocated could be greater
+ * than what has been asked in order to accommodate fixed memory allocations. If there is not a sufficient memory
  * available, it returns a @code nullptr @endcode
  *
- * If an array of Objects is need then provide malloc with a number to specify how many objects to
- * create and it will create that many objects.
+ * This is only used for creating single block of memory and not for array
  *
- * @pre this can also act a @code new @endcode keyword to create objects so there is no need to use
- * new keyword to create an object
- *
- * @pre malloc creates number of blocks of given type and it does not allocate memory by proving it
+ * @pre malloc creates number of blocks of given type and it does not allocate memory by providing it
  * number of bytes
  *
- * @tparam Type pointer type
- * @param num number of blocks of size @p Type
+ * @tparam T pointer type
+ * @tparam Args Argument types for non-fundamental Types
+ * @param args the arguments for @code Args @end types
  * @return address to memory allocated
  */
-template<typename Type>
-Type *malloc(wlp::size_type num = 1) {
-    void *memory = nullptr;
+template<
+        typename T,
+        typename... Args,
+        typename = typename wlp::enable_if<!wlp::is_array<T>::value, bool>::type>
+T *malloc(Args... args) {

```

Comment:
I think it might be better to use `is_fundamental` to overload a different function, that way you reduce having the if branch. i.e. `enable_if<is_fundamental<T>::value && !is_array<T>::value>`, `enable_if<!is_fundamental<T>::value && !is_array<T>::value>` ... etc,

might be a bit tedious but it's fine if the code has a lot of copy paste

Reply:
also can leave this for a later hot fix

Reply:
I can add the quick fixup later on as a HF